backend/tests/v5/e2e/routes/users.test.js

```diff
@@ -0,0 +1,150 @@
+/**
+ *  Copyright (C) 2021 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const SuperTest = require('supertest');
+const ServiceHelper = require('../../helper/services');
+const { src } = require('../../helper/path');
+const session = require('supertest-session');
+
+const { templates } = require(`${src}/utils/responseCodes`);
+
+let testSession;
+let server;
+let agent;
+
+// This is the user being used for tests
+const testUser = ServiceHelper.generateUserCredentials();
+const lockedUser = ServiceHelper.generateUserCredentials();
+const lockedUserWithExpiredLock = ServiceHelper.generateUserCredentials();
+const userWithFailedAttempts = ServiceHelper.generateUserCredentials();
+const userEmail = 'example@email.com';
+
+const setupData = async () => {
+	await Promise.all([
+		ServiceHelper.db.createUser(testUser, [], { email: userEmail }),
+		ServiceHelper.db.createUser(lockedUser, [], { loginInfo: { failedLoginCount: 10,
+			lastFailedLoginAt: new Date() } }),
+		ServiceHelper.db.createUser(lockedUserWithExpiredLock, [], { loginInfo: { failedLoginCount: 10,
+			lastFailedLoginAt: new Date('1/1/18') } }),
+		ServiceHelper.db.createUser(userWithFailedAttempts, [], { loginInfo: { failedLoginCount: 6,
+			lastFailedLoginAt: new Date() } }),
+	]);
+};
+
+const testLogin = () => {
+	describe('Login user', () => {
+		test('should log in a user using username', async () => {
+			await agent.post('/v5/login/')
+				.send({ user: testUser.user, password: testUser.password })
+				.expect(templates.ok.status);
+		});
+
+		test('should log in a user using email', async () => {
+			await agent.post('/v5/login/')
+				.send({ user: userEmail, password: testUser.password })
+				.expect(templates.ok.status);
+		});
+
+		test('should fail with an incorrect username', async () => {
+			const res = await agent.post('/v5/login/')
+				.send({ user: 'randomUsername', password: testUser.password })
+				.expect(templates.incorrectUsernameOrPassword.status);
+			expect(res.body.code).toEqual(templates.incorrectUsernameOrPassword.code);
+		});
+
```

Comment:
can we add 2 more tests here:
- add an invalidArgument check (just so we're sure we're calling the function to check it)
- User attempting to log in when they're already logged in (should throw alreadyLoggedIn)